### ISSUE TITLE
OBSOLETE - [HOWTO] Automatic transfer of packs

### ISSUE THREAD

**Issue by emedvedev** [StackStorm-Exchange/exchange-incubator#7]
Once a pull request is reviewed and merged, the following needs to happen:

1. An empty repository should be bootstrapped in the StackStorm-Exchange org with [exchange-bootstrap.sh](https://github.com/StackStorm-Exchange/ci/blob/master/utils/exchange-bootstrap.sh).

2. The submitted pack should be transferred into that repository in one of the two ways:

* as a new commit, if the pack has been submitted as a dir;
* as a merge with `--allow-unrelated-histories` if the pack has been submitted as a submodule and has git history.
 
3. The submitted pack should be removed from this repository once the transfer is complete and verified (automatically).

`exchange-bootstrap.sh` takes care of the CI, builds, and index updates, so the pack should automatically get to the Exchange index after the correct transfer, given it doesn't have linting errors (#6).

**Comment by emedvedev** [comment]
To give a complete example, here's how to transfer https://github.com/StackStorm/st2-mysql to the Exchange:

1. Run the bootstrap script with the necessary env variables:

```
$ export SLACK_WEBHOOK_URL=XXX
$ export CIRCLECI_TOKEN=XXX
$ export USERNAME=XXX
$ export PASSWORD=XXX
$ ./exchange-bootstrap.sh mysql
```

This will create the [StackStorm-Exchange/stackstorm-mysql](https://github.com/StackStorm-Exchange/stackstorm-mysql) repository and configure CircleCI for it.

2. Clone the Exchange repository, add the source repository as a remote, and create a branch for the transfer:

```
$ git clone https://github.com/stackstorm-exchange/stackstorm-mysql
$ git remote add source https://github.com/stackstorm/st2-mysql
$ cd stackstorm-mysql/
$ git checkout -b transfer
```

3. Pull to the `transfer` branch from the `master` branch of the source and resolve merge conflicts if they appear:

```
$ git pull source master --allow-unrelated-histories
// Automatic merge failed because of .gitignore; fix conflicts and then commit the result.
$ vim .gitignore
$ git add .gitignore
$ git commit -am 'Transfer the pack from StackStorm/st2-mysql'
```

4. Push to GitHub:

```
git push origin transfer
```

5. After the branch is uploaded to GitHub, create a pull request and fix linting errors (there _will be_ linting errors): 

https://github.com/StackStorm-Exchange/stackstorm-mysql/pull/1

6. Merge. You are awesome!

**Comment by LindsayHill** [comment]
Can you check that exchange-bootstrap is setting up CircleCI properly? Defaults seem to be *not* to run builds for pull requests from forks - e.g. check the new stackstorm-mysql settings https://circleci.com/gh/StackStorm-Exchange/stackstorm-mysql/edit#advanced-settings

**Comment by emedvedev** [comment]
@LindsayHill true, looks like it needs to be set manually. I'll see if I can modify the script to do that.

**Comment by LindsayHill** [comment]
For others watching this, I have fixed the script to now configure CircleCI to run builds for PRs from forks

**Comment by cognifloyd** [comment]
I edited the previous instructions to show the current required variables for the `exchange-bootstrap.sh` script.

WIP below this point...

If someone submits a pack as a submodule to the exchange-incubator, use the instructions above: https://github.com/StackStorm-Exchange/exchange-incubator/issues/7#issuecomment-281247786

If someone submits a pack as a subdirectory in a PR, then we need to use a slightly different process.

1. Bootstrap the exchange repo (see the other instructions for an example; this step is the same).
2.  Clone the Exchange repository, add the incubator repository as a remote:
```
$ git clone https://github.com/StackStorm-Exchange/stackstorm-mysql
$ cd stackstorm-mysql/
$ git remote add incubator https://github.com/StackStorm-Exchange/exchange-incubator
```
3. Fetch the pull request (use the correct PR number! 7 is the number of this issue.):
```
$ git fetch incubator pull/7/head
```

Here, the instructions may vary slightly.

4. When there is only one commit, cherry-pick, resolve conflicts, and amend the commit:
```
$ git cherry-pick 989fe28cd05df43bf0b03639eca1624dabc7e7d7
$ git mv mysql/* ./
# drop unnecessary execute bits
$ chmod -x *.md *.yaml *.example *.txt *.png
# fix any conflicts
$ git diff
$ vim .gitignore
$ git add .gitignore
$ git commit --amend
# add a note saying "Transferred from: <URL to incubator PR>"
```

**Comment by cognifloyd** [comment]
Now that we have a bootstrap pack workflow (triggered with `!bootstrap pack` comment on an approved incubator PR) added in #172, the instructions here are out-of-date. So, I'm closing this.

Note: the logic in #172 came in part from these instructions. But now it's a lot less tedious :wink:.

**Comment by armab** [comment]
We need just very minimal instructions about the new behavior.

Adding the `!bootstrap pack` step in the https://github.com/StackStorm-Exchange/exchange-incubator README would be sufficient to replace this issue with something.
